pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -107,11 +108,18 @@ type SchedulingQueue interface {
 	// Pop removes the head of the queue and returns it. It blocks if the
 	// queue is empty and waits until a new item is added to the queue.
 	Pop() (*framework.QueuedPodInfo, error)
+	// Done must be called pod returned by Pop. This allows the queue to
+	// keep track of which pods are currently being processed.
+	Done(types.UID)
```

Comment:
This is a very good idea!
Could you pull this into an independent PR?

There might be some race conditions if you are calling this before Requeue. Or maybe Requeue implicitly calls Done?

Reply:
(Actually, the inflight Pod idea is from @pohly 😁 )

Ack. Let me rethink about that.

Reply:
I double-checked but the current code should always call Done() after executing Requeues for in-flight Pods.

- Done() is called at the end of the scheduling cycle. ([in `defer` of `scheduleOne`](https://github.com/sanposhiho/kubernetes/blob/poc-wise-enqueueing/pkg/scheduler/schedule_one.go#L76))
- Requeue for in-flight Pods are called in [sched.FailureHandler](https://github.com/sanposhiho/kubernetes/blob/poc-wise-enqueueing/pkg/scheduler/schedule_one.go#L107).
  - `sched.FailureHandler` calles `AddUnschedulableIfNotPresent`, and `AddUnschedulableIfNotPresent` calls Requeues for in-flight Pod.
    -  https://github.com/sanposhiho/kubernetes/blob/poc-wise-enqueueing/pkg/scheduler/internal/queue/scheduling_queue.go#L622